.github/workflows/test-plugins-against-cockpit.yml

```diff
@@ -0,0 +1,188 @@
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          cache: 'npm'
+
+      - name: Extract Plugins Version
+        id: extract-plugins-version
+        run: |
+          plugins_version=$(jq -r '.version' package.json)
+          echo "plugins_version=${plugins_version}" >> $GITHUB_ENV
+          echo "::set-output name=plugins_version::${plugins_version}"
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build
+        run: npm run build
+
+      - name: Upload build artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: build
+          if-no-files-found: error
+          retention-days: 5
+          path: |
+            dist/sag-pkg-community-plugins/**
+
+  run-tests-against-shell:
+    needs: [collect-shell-versions, build-plugins]
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        version_data: ${{ fromJson(needs.collect-shell-versions.outputs.shell_versions) }}
+    env:
+      JSON: ${{ toJson(matrix.version_data) }}
+      VERSION: ${{ matrix.version_data.version }}
+      MAJOR: ${{ matrix.version_data.major }}
+      TAG: ${{ matrix.version_data.tag }}
+      PLUGINS_VERSION: ${{ needs.build-plugins.outputs.plugins_version }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Retrieve cached JSON files
+        if: ${{ github.event_name == 'schedule' }}
+        id: retrieve-cache
+        uses: actions/cache@v4
+        with:
+          path: ./cache-files
+          key: cache-json-files-${{ runner.os }}-${{ env.TAG }}-${{ env.PLUGINS_VERSION }}
+          restore-keys: |
+            cache-json-files-${{ runner.os }}-${{ env.TAG }}-${{ env.PLUGINS_VERSION }}
+
+      - name: Check if version is already tested
+        if: ${{ github.event_name == 'schedule' }}
+        id: check-version
+        run: |
+          mkdir -p ./cache-files
+          file="./cache-files/${{ env.TAG }}-${{ env.PLUGINS_VERSION }}.json"
+          if [ -f "$file" ]; then
+            echo "Cache file exists: $file"
+            alreadyTestedShellVersion=$(jq -r '.shellVersion' $file)
+            alreadyTestedPluginsVersion=$(jq -r '.pluginsVersion' $file)
+            if [[ "$alreadyTestedShellVersion" == "${{ env.VERSION }}" && "$alreadyTestedPluginsVersion" == "${{ env.PLUGINS_VERSION }}" ]]; then
+              echo "Shell version $alreadyTestedShellVersion with plugins of version $alreadyTestedPluginsVersion already tested."
+              echo "skip_job=true" >> $GITHUB_ENV
+            fi
+          else
+            echo "No cache file found for ${{ env.TAG }} with plugins version ${{ env.PLUGINS_VERSION }}"
+          fi
```

Comment:
Would move this step out of here, so you would not actually need to start a job for a version that was already tested..
Maybe you could move it into the `collect-shell-versions` job?

Reply:
that's good idea. I can also skip building plugin if no version needs to be test against, this will reduce redundant jobs significantly.